src/NLog/LayoutRenderers/ProcessInfoLayoutRenderer.cs

```diff
@@ -107,7 +113,11 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
             if (this.propertyInfo != null)
             {
                 var formatProvider = GetFormatProvider(logEvent);
-                builder.Append(Convert.ToString(this.propertyInfo.GetValue(this.process, null), formatProvider));
+                var value = this.propertyInfo.GetValue(this.process, null);
+                if (value != null && this.Format != null)
+                    builder.Append(Internal.FormatHelper.ToStringWithOptionalFormat(value, this.Format, formatProvider));
```

Comment:
This method already handles null for both.  ;)

(and it's an extension method)

So my proposal: 

```
 builder.Append(value.ToStringWithOptionalFormat(this.Format, formatProvider));

```